server/bin/pbench-index.py

```diff
@@ -299,26 +299,29 @@ def main(options, name):
                 except UnsupportedTarballFormat as e:
                     idxctx.logger.warning("Unsupported tar ball format: {}", e)
                     tb_res = 4
+                    indexing_failure(os.path.realpath(tb),"UnsupportedTarballFormat", tb_res)
```

Comment:
There should probably be a single call to `indexing_failure()` in a `finally` section: all we need to do is save `e` to a more permanent place (`saved_e` perhaps - initialized to `None`). Then in the `finally` section, we check `saved_e` and if not `None`, we call `indexing_failure()`.